src/storm/generator/PrismNextStateGenerator.cpp

```diff
@@ -159,58 +160,47 @@ namespace storm {
 
             // If all states are initial, we can simplify the enumeration substantially.
             if (program.hasInitialConstruct() && program.getInitialConstruct().getInitialStatesExpression().isTrue()) {
-                CompressedState initialState(this->variableInformation.getTotalBitOffset(true));
-
-                std::vector<int_fast64_t> currentIntegerValues;
-                currentIntegerValues.reserve(this->variableInformation.integerVariables.size());
-                for (auto const& variable : this->variableInformation.integerVariables) {
-                    STORM_LOG_THROW(variable.lowerBound <= variable.upperBound, storm::exceptions::InvalidArgumentException, "Expecting variable with non-empty set of possible values.");
-                    currentIntegerValues.emplace_back(0);
-                    initialState.setFromInt(variable.bitOffset, variable.bitWidth, 0);
+                
+                // Create vectors holding all possible values
+                std::vector<std::vector<uint64_t>> allValues;
+                for (auto const& intVar : this->variableInformation.integerVariables) {
+                    STORM_LOG_ASSERT(intVar.lowerBound <= intVar.upperBound, "Expecting variable with non-empty set of possible values.");
+                    // The value of integer variables is shifted so that 0 is always the smallest possible value
+                    allValues.push_back(storm::utility::vector::buildVectorForRange<uint64_t>(static_cast<uint64_t>(0), intVar.upperBound + 1 - intVar.lowerBound));
```

Comment:
Similar to the JANI case: Could it make sense to only store the bounds? in particular, if I see it correctly here all lower range bounds are 0, so it would suffice to store the upper bound.

Reply:
Here I agree that something like [counting_range](https://www.boost.org/doc/libs/1_76_0/libs/range/doc/html/range/reference/ranges/counting_range.html) would be better. However, I still think this is not very concerning because actually exploring those states is much much more costly

Reply:
Yep, I agree.